File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.java
class Solution {
    public int partitionString(String s) {
        
        int []arr=new int[26];Arrays.fill(arr,0);
        
        int lastIdx=0,cnt=0;
        for(int i=0;i<s.length();i++){
            if(arr[s.charAt(i)-'a']>=lastIdx){
                cnt++;
                lastIdx=i+1;
            }
            arr[s.charAt(i)-'a']=i+1;
        }
        
        return cnt;
    }
}

File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.cpp
class Solution {
public:
    int partitionString(string s) {
        unordered_map<char,int>map;
        int cnt=0;
        for(char c:s){
            if(map[c]!=0){
                cnt++;
                map.clear();
            }
            map[c]++;
        }
        return cnt+1;
    }
};